client/etcd/client.go

```diff
@@ -81,25 +84,24 @@ type csclient struct {
 	clis map[string]*clientv3.Client
 
 	opts    Options
+	sdOpts  sdclient.Options
 	kvScope tally.Scope
 	sdScope tally.Scope
 	hbScope tally.Scope
 	logger  xlog.Logger
 	newFn   newClientFn
 
-	sdOnce sync.Once
-	sd     services.Services
-	sdErr  error
+	defaultServicesOnce sync.Once
+	defaultServices     services.Services
+	defaultServicesErr  error
 
 	txnOnce sync.Once
 	txn     kv.TxnStore
 	txnErr  error
 }
 
-func (c *csclient) Services() (services.Services, error) {
-	c.createServices()
-
-	return c.sd, c.sdErr
+func (c *csclient) Services(opts services.Options) (services.Services, error) {
```

Comment:
Do you want to do sth like
```
if opts == nil {
  opts = services.NewOptions()
}
...
```
so users of default services can just pass in a nil?